yt_dlp/extractor/redgifs.py

```diff
@@ -80,15 +75,110 @@ def _real_extract(self, url):
+            'categories': gif_data.get('tags') or [],
+            'webpage_url': webpage_url,
+            'tags': gif_data.get('tags'),
             'age_limit': 18,
             'formats': formats,
         }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url).lower()
+
+        video_info = self._download_json(
+            'https://api.redgifs.com/v2/gifs/%s' % video_id,
+            video_id, 'Downloading video info')
+        if 'error' in video_info:
+            raise ExtractorError(f'RedGifs said: {video_info["error"]}', expected=True)
+
+        return self._parse_gif_data(video_info['gif'])
+
+
+class RedGifsSearchIE(RedGifsIE):
+    IE_DESC = 'Redgifs search'
+    _VALID_URL = r'https?://(?:www\.)?redgifs\.com/browse\?(?P<query>.*)'
+    _PAGE_SIZE = 80
+    _TESTS = [
+        {
+            'url': 'https://www.redgifs.com/browse?tags=Lesbian',
+            'info_dict': {
+                'id': 'tags=Lesbian',
+                'title': 'Lesbian',
+                'description': 'RedGifs search for Lesbian, ordered by trending'
+            },
+            'playlist_mincount': 100,
+        },
+        {
+            'url': 'https://www.redgifs.com/browse?type=g&order=latest&tags=Lesbian',
+            'info_dict': {
+                'id': 'type=g&order=latest&tags=Lesbian',
+                'title': 'Lesbian',
+                'description': 'RedGifs search for Lesbian, ordered by latest'
+            },
+            'playlist_mincount': 100,
+        },
+        {
+            'url': 'https://www.redgifs.com/browse?type=g&order=latest&tags=Lesbian&page=2',
+            'info_dict': {
+                'id': 'type=g&order=latest&tags=Lesbian&page=2',
+                'title': 'Lesbian',
+                'description': 'RedGifs search for Lesbian, ordered by latest'
+            },
+            'playlist_count': 80,
+        }
+    ]
+
+    def _fetch_page(self, video_id, api_query, page=1):
+        api_query['page'] = page
+        data = self._download_json(
+            'https://api.redgifs.com/v2/gifs/search',
+            video_id,
+            query=api_query
+        )
+        if 'error' in data:
+            raise ExtractorError(f'RedGifs said: {data["error"]}', expected=True)
+
+        return [self._parse_gif_data(entry) for entry in data['gifs']]
+
+    def _real_extract(self, url):
+        query_str = self._match_valid_url(url).group('query')
+
+        query = compat_parse_qs(query_str)
+        if not query.get('tags'):
+            raise ExtractorError('Invalid query tags', expected=True)
+
+        tags = query.get('tags')[0]
+        order = query.get('order', ('trending',))[0]
+        api_query = {
+            'search_text': tags,
+            'order': order,
+        }
+        if query.get('type'):
+            api_query['type'] = query.get('type')[0]
+
+        if query.get('page'):
+            page = query.get('page', (1,))[0]
```

Comment:
Since its inside if, no default is needed
```suggestion
            page = query.get('page')[0]
```